### PR TITLE
(PDB-2309) Document historical-catalogs views in PE

### PR DESCRIPTION
**Pull request by ajroetker**
This commit adds documentation for the PE only historical-catalogs
views.

### CODE REVIEW

**Comment by ajroetker** [comment]
Disregard tests, this is a docs only change

File: documentation/api/ext/v1/resource-graphs.markdown [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
+title: "PuppetDB 3.2 » Extensions API (PE only)"
```
Comment (mullr) [inline documentation/api/ext/v1/resource-graphs.markdown]: somehow I thought we were keeping this in the pe repo.

File: documentation/api/ext/v1/resource-graphs.markdown [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+You can query resource-graphs by making an HTTP request to the
+`/pdb/ext/v1/resource-graphs` endpoint.
+
+## `/pdb/ext/v1/resource-graphs`
+
+This will return a JSON array containing all the resource-graphs for each node
+in your infrastructure.
+
+This resource-graph is a report and a catalog munged together so that all
```
Comment (mullr) [inline documentation/api/ext/v1/resource-graphs.markdown]: munge, is that a technical term?

Comment (mullr) [inline documentation/api/ext/v1/resource-graphs.markdown]: Do we want to call it a "resouce-graph"? That's a little strange. I don't understand how this is a graph at all actually. 'combined-resources'?

Comment (ajroetker) [inline documentation/api/ext/v1/resource-graphs.markdown]: It's just what the view is like in the console, I'm open to different names

File: documentation/api/ext/v1/resource-graphs.markdown (hunk 1 above)
Comment (ajroetker) [inline documentation/api/ext/v1/resource-graphs.markdown]: Nope, we are not, all the PE docs should be in the FOSS repo afiak now

File: documentation/api/ext/v1/resource-graphs.markdown (hunk 2 above)
Comment (senior) [inline documentation/api/ext/v1/resource-graphs.markdown]: How about "This resource-graph is a unified view of data from the report and the catalog..."

File: documentation/api/ext/v1/resource-graphs.markdown [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+### Query Fields
+
+* `certname` (string): the certname associated with the resource-graph
+* `environment` (string): the environment assigned to the node that submitted
+  the report.
+* `transaction_uuid` (string): string used to identify a puppet run.
+* `catalog_uuid` (string): a string used to tie a catalog to its associated
+  reports
+* `code_id` (string): a string used to tie a catalog to the Puppet code which
```
Comment (senior) [inline documentation/api/ext/v1/resource-graphs.markdown]: I think we need (maybe not here) more detailed information about this and the catalog_uuid. Maybe we can link to their definitions in the Puppet docs? @joshcooper might be able to point us in the right direction.

File: documentation/api/ext/v1/resource-graphs.markdown [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    [
+      {
+        "catalog_uuid" : "53b72442-3b73-11e3-94a8-1b34ef7fdc95",
+        "certname": "host-1",
+        "code_id": null,
+        "environment": "production",
+        "noop": false,
+        "producer_timestamp": "2016-01-07T20:40:21.119Z",
+        "status": "changed",
+        "transaction_uuid": "2211965a-826d-4bb7-a8ab-9eef8c0c43d1"
```
Comment (senior) [inline documentation/api/ext/v1/resource-graphs.markdown]: Missing a comma

File: documentation/api/query/v4/catalogs.markdown [hunk 5]
```diff
@@ -207,11 +218,42 @@ deactivated or expired.
...
+## `/pdb/ext/v1/historical-catalogs` (PE only)
+
+This will return a JSON array containing the catalogs for each node in your
+infrastructure, not just the most recent. By default this will include the most
+recent 14 days worth of catalogs for each node.
+
+**Note on querying resources and edges**
+
+The `historical-catalogs` endpoint does not support querying on the value of
+`resources` or `edges`. Making `resources` and `edges` queryable historically
```
Comment (senior) [inline documentation/api/query/v4/catalogs.markdown]: Seems like we can omit the future statement.

**Comment by senior** [comment]
Minor comments, other than that :+1:

File: documentation/api/ext/v1/resource-graphs.markdown (hunk 3 above)
Comment (ajroetker) [inline documentation/api/ext/v1/resource-graphs.markdown]: I couldn't find anything other than the release notes related to this (`code_id`) from poking around.

**Comment by ajroetker** [comment]
@senior @mullr comments should be all addressed

File: documentation/api/query/v4/catalogs.markdown [hunk 6]
```diff
@@ -207,11 +218,41 @@ deactivated or expired.
...
+This route is an extension of the [`/resources`][resources] endpoint. It uses
+the exact same parameters, operators, fields, and response format.
+
+If you provide a `query` parameter, it will specify additional criteria, which
+will be used to return a subset of the information normally returned by this
+route.
+
+## `/pdb/ext/v1/historical-catalogs` (PE only)
+
+This will return a JSON array containing the catalogs for each node in your
```
Comment (mullr) [inline documentation/api/query/v4/catalogs.markdown]: "containing all the stored catalogs"

**Comment by mullr** [comment]
just one nit, lgtm otherwise

**Comment by ajroetker** [comment]
@mullr update for you comment

File: documentation/api/ext/v1/resource-graphs.markdown (hunk 3 above)
Comment (senior) [inline documentation/api/ext/v1/resource-graphs.markdown]: alright, we'll revisit after the feature is in place on the puppet side

**Pull request merged**